src/EditorFeatures/CSharpTest/SpellCheck/SpellCheckSpanTests.cs

```diff
@@ -6,676 +6,682 @@
+            """");
+    }
+
+    [Fact]
+    public async Task TestString12()
+    {
+        await TestAsync(""""
+            """
+                goo
+                ""
+            """");
+    }
+
+    [Fact]
+    public async Task TestString13()
+    {
+        await TestAsync(""""
+            """
+                goo
+                "
+            """");
+    }
+
+    [Fact]
+    public async Task TestString14()
+    {
+        await TestAsync(""""
+            """
+                goo
+            """");
+    }
+
+    [Fact]
+    public async Task TestString15()
+    {
+        await TestAsync("""
+            $"{|String: goo |}"
+            """);
+    }
+
+    [Fact]
+    public async Task TestString16()
+    {
+        await TestAsync("""
+            $"{|String: goo |}{0}{|String: bar |}"
+            """);
+    }
+
+    [Fact]
+    public async Task TestString17()
+    {
+        await TestAsync(""""
+            $"""{|String: goo |}{0}{|String: bar |}"""
+            """");
+    }
+
+    [Fact]
+    public async Task TestString18()
+    {
+        await TestAsync(""""
+            $"""{|String: goo |}{0:abcd}{|String: bar |}"""
+            """");
+    }
+
+    [Fact]
+    public async Task TestEscapedString1()
+    {
+        await TestAsync(""" " {|String:goo|}\r\n{|String:bar |}" """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString2()
+    {
+        await TestAsync(""" " {|String:goo|}\r\t{|String:bar |}" """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString3()
+    {
+        await TestAsync(""" " {|String:goo|}\r\u0065bar " """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString4()
+    {
+        await TestAsync(""" " {|String:C|}:\t{|String:ests |}" """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString5()
+    {
+        await TestAsync(""" {|String:@" C:\tests "|} """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString6()
+    {
+        await TestAsync(""" " {|String:C|}:\\{|String:tests |}" """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString7()
+    {
+        await TestAsync(""" " {|String:C|}:\\{|String:tests|}\\{|String:goo |}" """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString8()
+    {
+        await TestAsync(""" {|String:@" C:\tests\goo "|} """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString9()
+    {
+        await TestAsync(""" $" {|String:C|}:\\{|String:tests|}\\{|String:goo |}{0} {|String:C|}:\\{|String:tests|}\\{|String:bar |}" """);
+    }
+
+    [Fact]
+    public async Task TestEscapedString10()
+    {
+        await TestAsync(""" $@"{|String: C:\tests\goo |}{0}{|String: C:\tests\bar |}" """);
+    }
+
+    [Fact, WorkItem("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1908466")]
+    public async Task TestEscapedString11()
+    {
+        await TestAsync(""" " {|String:open telemetry for audits |}\n\t {|String:and |}0{|String:Tel table doesn't exist yet |}" """);
```

Comment:
important change here is that `"doesn't"` isn't split into multiple words because of hte `'`.